### PR TITLE
Feature/add solution items folder and copy to plugins task

### PR DESCRIPTION
**Pull request by ErisApps**
This PR adds some useful tidbits for contributors/developers.
- Adds a solutionItems virtual folder which groups several markdown files as well as the .csproj.user and manifest.json
- Adds a CopyToPlugins task to the build process. Configurable using a flag.

### CODE REVIEW

File: BeatSaberHTTPStatus/BeatSaberHTTPStatusPlugin.csproj [hunk 1]
```diff
@@ -115,6 +116,13 @@
     <ReplaceFileText InputFilename="$(IntermediateSourcePath)manifest.json" OutputFilename="$(IntermediateSourcePath)manifest.json" MatchExpression="$SEMVER_VERSION$" ReplacementText="$(SemVerVersion)" />
     <ReplaceFileText InputFilename="$(IntermediateSourcePath)manifest.json" OutputFilename="$(IntermediateSourcePath)manifest.json" MatchExpression="$BS_VERSION$" ReplacementText="$(BeatSaberVersion)" />
   </Target>
+  <!-- Copy the plugin to the Beat Saber installation -->
+  <Target Name="CopyToPlugins" AfterTargets="AfterBuild" Condition=" '$(CopyToPlugins)' != '' And $(CopyToPlugins) ">
+    <Message Text="Copying $(OutDir)BeatSaberHTTPStatus.dll to Plugins folder" Importance="high"/>
+    <Copy SourceFiles="$(OutDir)BeatSaberHTTPStatus.dll" DestinationFiles="$(GameDirPath)Plugins\BeatSaberHTTPStatus.dll"/>
+    <Message Text="Copying $(OutDir)websocket-sharp.dll to Beat Saber_Data folder" Importance="high"/>
+    <Copy SourceFiles="$(OutDir)BeatSaberHTTPStatus.dll" DestinationFiles="$(GameDirPath)Beat Saber_Data\Managed\websocket-sharp.dll"/>
```
Comment (opl-) [inline BeatSaberHTTPStatus/BeatSaberHTTPStatusPlugin.csproj:124]: :thinking:
```suggestion
    <Copy SourceFiles="$(OutDir)websocket-sharp.dll" DestinationFiles="$(GameDirPath)Beat Saber_Data\Managed\websocket-sharp.dll"/>
```

File: BeatSaberHTTPStatusPlugin.sln [hunk 2]
```diff
@@ -12,4 +25,10 @@ Global
 		{1C0571AC-F8D6-4F9E-93B2-2F23892AE61E}.Release|Any CPU.ActiveCfg = Release|Any CPU
 		{1C0571AC-F8D6-4F9E-93B2-2F23892AE61E}.Release|Any CPU.Build.0 = Release|Any CPU
 	EndGlobalSection
+	GlobalSection(SolutionProperties) = preSolution
+			HideSolutionNode = FALSE
+		EndGlobalSection
+		GlobalSection(ExtensibilityGlobals) = postSolution
+			SolutionGuid = {F5FB7413-F469-4EC8-BB25-F2574E515169}
+		EndGlobalSection
```
Comment (opl-) [inline BeatSaberHTTPStatusPlugin.sln:33]: Indentation.

File: BeatSaberHTTPStatusPlugin.sln [hunk 3]
```diff
@@ -1,6 +1,19 @@
+
 Microsoft Visual Studio Solution File, Format Version 12.00
+# Visual Studio Version 16
+VisualStudioVersion = 16.0.30011.22
+MinimumVisualStudioVersion = 10.0.40219.1
 Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "BeatSaberHTTPStatus", "BeatSaberHTTPStatus\BeatSaberHTTPStatusPlugin.csproj", "{1C0571AC-F8D6-4F9E-93B2-2F23892AE61E}"
 EndProject
+Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Solution items", "Solution items", "{2623DD3B-3EC1-43FE-AD24-5D31D1E49F28}"
```
Comment (opl-) [inline BeatSaberHTTPStatusPlugin.sln:8]: Is the "Solution items" a special string or could it be renamed to something like "Useful files" or "LOOK AT THESE"?

File: BeatSaberHTTPStatus/BeatSaberHTTPStatusPlugin.csproj (hunk 1 above)
Comment (ErisApps) [reply BeatSaberHTTPStatus/BeatSaberHTTPStatusPlugin.csproj]: How did I even miss that 😓

File: BeatSaberHTTPStatusPlugin.sln (hunk 3 above)
Comment (ErisApps) [reply BeatSaberHTTPStatusPlugin.sln:8]: It's just a normal string so it's perfectly possible to rename it to something else.
I think you'll need the 2 strings in sync though.

Comment (ErisApps) [reply BeatSaberHTTPStatusPlugin.sln]: Renamed to "Useful files" as per conversation.

**Comment by opl-** [comment]
:ok_hand:

**Pull request merged**